frontend/src/metabase/components/EmptyState/EmptyState.tsx

```diff
@@ -0,0 +1,107 @@
+import Button from "metabase/core/components/Button";
+import type { IconName } from "metabase/ui";
+import { Icon, isValidIconName } from "metabase/ui";
+import Link from "metabase/core/components/Link";
+import Text from "metabase/components/type/Text";
+import {
+  EmptyStateActions,
+  EmptyStateFooter,
+  EmptyStateHeader,
+  EmptyStateIllustration,
+} from "./EmptyState.styled";
+
+// Don't break existing empty states
+// TODO - remove these and update empty states with proper usage of illustrationElement
+const LegacyIcon = ({ icon }: { icon: IconName }) =>
+  icon ? <Icon name={icon} className="text-light" size={40} /> : null;
+
+const LegacyImage = ({
+  image,
+  imageHeight,
+  imageClassName,
+  message,
+}: {
+  image: string;
+  imageHeight?: number;
+  imageClassName?: string;
+  message?: string;
+}) =>
+  image ? (
+    <img
+      src={`${image}.png`}
+      width="300px"
+      height={imageHeight}
+      alt={message}
+      srcSet={`${image}@2x.png 2x`}
+      className={imageClassName}
+    />
+  ) : null;
+
+type EmptyStateProps = {
+  message?: React.ReactNode;
+  title?: React.ReactNode;
+  action?: React.ReactNode;
+  link?: string;
+  illustrationElement?: React.ReactNode;
+  onActionClick?: () => void;
+  className?: string;
+  icon?: IconName;
+  image?: string;
+};
+
+const EmptyState = ({
+  title,
+  message,
+  action,
+  link,
+  illustrationElement,
+  onActionClick,
+  className,
+  icon,
+  image,
+  ...rest
+}: EmptyStateProps) => (
+  <div className={className}>
+    <EmptyStateHeader>
+      {illustrationElement && (
+        <EmptyStateIllustration className="empty-state-illustration">
+          {illustrationElement}
+        </EmptyStateIllustration>
+      )}
+      <div>
+        {isValidIconName(icon) && <LegacyIcon icon={icon} {...rest} />}
```

Comment:
I added a more rigorous check of the icon name, which typescript needs

It turns out that there is at least one icon with the name set to "all", which is not a valid icon name
https://github.com/metabase/metabase/blob/ea7f5b7e443cd4f1cd6a725999c82b53bef87022/frontend/src/metabase/reference/metrics/MetricQuestions.jsx#L37